src/main/python/systemds/script_building/script.py

```diff
@@ -79,7 +79,34 @@ def execute(self) -> JavaObject:
                                                             _list_to_java_array([self.out_var_name]))
             for (name, input_node) in self.inputs.items():
                 input_node.pass_python_data_to_prepared_script(gateway.jvm, name, self.prepared_script)
-        return self.prepared_script.executeScript()
+
+            if lineage:
+                connection.setLineage(True)
+
+        ret = self.prepared_script.executeScript()
+        if lineage:
+            print(self.prepared_script.getLineageTrace(self.out_var_name))
```

Comment:
Maybe we could return this instead so the user can decide if he wants to print it or maybe write it to a file or do something else with it.

Reply:
That would also allow us to better test it

Reply:
It didn't look good to me that compile() is returning the trace, that is why I provided the other separate function getLineageTrace() to get the lineage trace of any intermediates.

Reply:
one of the strengths of python is that you are able to return multiple things, so in this case you could just return it, and on the receiving end (where the method is called) it can be filtered out.